Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/create/TraceLocationCreateFragment.kt

```diff
@@ -0,0 +1,126 @@
+
+    private val binding: TraceLocationCreateFragmentBinding by viewBindingLazy()
+    private val navArgs by navArgs<TraceLocationCreateFragmentArgs>()
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: TraceLocationCreateViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TraceLocationCreateViewModel.Factory
+            factory.create(navArgs.category)
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.toolbar.setNavigationOnClickListener { popBackStack() }
+
+        viewModel.uiState.observe(viewLifecycleOwner) { state ->
+            binding.toolbar.setSubtitle(state.title)
+            binding.valueStart.text = state.getStartDate(requireContext().getLocale())
+            binding.valueEnd.text = state.getEndDate(requireContext().getLocale())
+            binding.layoutBeginn.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
+            binding.layoutEnd.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
+            binding.valueLengthOfStay.text = state.getLength(resources)
+            binding.buttonSubmit.isEnabled = state.isSendEnable
+        }
+
+        binding.descriptionInputEdit.doOnTextChanged { text, _, _, _ ->
+            viewModel.description = text?.toString()
+        }
+
+        binding.placeInputEdit.doOnTextChanged { text, _, _, _ ->
+            viewModel.address = text?.toString()
+        }
+
+        binding.layoutBeginn.setOnClickListener {
+            it.hideKeyboard()
+            showDatePicker(viewModel.start) { value ->
+                viewModel.start = value
+            }
+        }
+
+        binding.layoutEnd.setOnClickListener {
+            it.hideKeyboard()
+            showDatePicker(viewModel.end) { value ->
+                viewModel.end = value
+            }
+        }
+
+        binding.layoutLengthOfStay.setOnClickListener {
+            it.hideKeyboard()
+            showDurationPicker()
+        }
+
+        binding.buttonSubmit.setOnClickListener {
+            viewModel.send()
+        }
+    }
+
+    private fun showDatePicker(defaultValue: LocalDateTime?, callback: (LocalDateTime) -> Unit) {
+        MaterialDatePicker.Builder.datePicker().setSelection(defaultValue?.toDateTime(DateTimeZone.UTC)?.millis).build()
+            .apply {
+                addOnPositiveButtonClickListener {
+                    showTimePicker(LocalDate(it), defaultValue?.hourOfDay, defaultValue?.minuteOfHour, callback)
+                }
+            }.show(childFragmentManager, DATE_PICKER_TAG)
+    }
+
+    private fun showTimePicker(date: LocalDate, hours: Int?, minutes: Int?, callback: (LocalDateTime) -> Unit) {
```

Comment:
Can we somehow set the locale for the Time Picker. I have set German as the language and it shows me AM/PM when selecting the time, which is not so nice.

Reply:
updated